x-pack/winlogbeat/module/testing.go

```
@@ -0,0 +1,331 @@
+		URL:              host,
+		Username:         user,
+		Password:         pass,
+		CompressionLevel: 3,
+		Transport:        httpcommon.HTTPTransportSettings{Timeout: time.Minute},
+	})
+	if err != nil {
+		t.Fatalf("unexpected error making connection: %v", err)
+	}
+	defer conn.Close()
+
+	err = conn.Connect()
+	if err != nil {
+		t.Fatalf("unexpected error making connection: %v", err)
+	}
+
+	info := beat.Info{
+		IndexPrefix: indexPrefix,
+		Version:     version.GetDefaultVersion(),
+	}
+	loaded, err := module.UploadPipelines(info, conn, true)
+	if err != nil {
+		t.Errorf("unexpected error uploading pipelines: %v", err)
+	}
+	wantPipelines := []string{
+		"powershell",
+		"powershell_operational",
+		"routing",
+		"security",
+		"sysmon",
+	}
+	if len(loaded) != len(wantPipelines) {
+		t.Fatalf("unexpected number of loaded pipelines: got:%d want:%d", len(loaded), len(wantPipelines))
+	}
+	want := regexp.MustCompile(`^` + indexPrefix + `-.*-(` + strings.Join(wantPipelines, "|") + `)$`)
+	pipelines := make(map[string]string)
+	for _, p := range loaded {
+		m := want.FindAllStringSubmatch(p, -1)
+		pipelines[m[0][1]] = p
+	}
+	_, ok := pipelines[pipeline]
+	if !ok {
+		t.Fatalf("failed to upload %q", pipeline)
+	}
+
+	cases, err := wintest.SimulatePipeline(host, user, pass, pipelines[pipeline], paths)
+	if err != nil {
+		t.Fatalf("unexpected error running simulate: %v", err)
+	}
+	for _, k := range cases {
+		name := filepath.Base(k.Path)
+		t.Run(name, func(t *testing.T) {
+			if k.Err != nil {
+				t.Errorf("unexpected error: %v", k.Err)
+			}
+
+			var events []mapstr.M
+			//nolint:errcheck // All the errors returned here are from mapstr.M queries and may be ignored.
+			for i, p := range k.Processed {
+				err = wintest.ErrorMessage(p)
+				if err != nil {
+					t.Errorf("unexpected ingest error for event %d: %v", i, err)
+				}
+
+				var event mapstr.M
+				err = json.Unmarshal(p, &event)
```

I think this error is not checked.

---

Thanks; fixed. It should never fail since the message has already been through a number of roundtrips, but better to be correct.